modular_skyrat/modules/vore/code/obj/belly.dm

```diff
@@ -0,0 +1,304 @@
+			living_gone.client.prefs.vr_prefs.update_static_data(living_gone)
+	if (iscarbon(gone))
+		var/mob/living/carbon/carbon_gone = gone
+		carbon_gone.update_suit_sensors()
+	check_mode()
+	update_static_vore_data()
+
+/obj/vbelly/proc/update_static_vore_data(force=FALSE, only_contents=FALSE, timer = FALSE)
+	static_data_cooldown = (force || timer) ? 0 : static_data_cooldown
+	static_timer = timer ? FALSE : static_timer
+	if (static_data_cooldown > world.time)
+		if (!static_timer)
+			static_timer = TRUE
+			addtimer(CALLBACK(src, .proc/update_static_vore_data, FALSE, FALSE, TRUE), static_data_cooldown - world.time)
+		return
+	static_data_cooldown = world.time + 0.5 SECONDS
+	for (var/mob/living/living in src)
+		if (living.client?.prefs?.vr_prefs)
+			living.client.prefs.vr_prefs.needs_update |= UPDATE_INSIDE
+			living.client.prefs.vr_prefs.update_static_data(living)
+	if (!only_contents && owner.client?.prefs?.vr_prefs)
+		owner.client.prefs.vr_prefs.needs_update |= UPDATE_CONTENTS
+		owner.client.prefs.vr_prefs.update_static_data(owner)
+
+/obj/vbelly/proc/get_belly_contents(ref=FALSE, living=FALSE, as_string=FALSE, ignored=null, full=FALSE)
+	var/list/belly_contents = list()
+	var/list/keep_track = list()
+	for (var/atom/movable/AM as anything in src)
+		if (ignored && (AM == ignored || (AM in ignored)))
+			continue
+		if (living && !isliving(AM))
+			continue
+		var/key = as_string ? "[AM][keep_track["[AM]"] ? " ([keep_track["[AM]"]])" : ""]" : AM
+		if (full)
+			belly_contents += list(list("name" = key, "absorbed" = (AM in absorbed), "ref" = ref(AM))) //double list because byond is dumb and this is the only way to add a list to a list afaik
+		else if (ref)
+			belly_contents[key] = ref(AM)
+		else
+			belly_contents += key
+		keep_track["[AM]"] = keep_track["[AM]"] ? keep_track["[AM]"] + 1 : 2
+	return belly_contents
+
+/obj/vbelly/proc/check_mode()
+	if (mode == VORE_MODE_HOLD)
+		return
+	START_PROCESSING(SSvore, src)
+
+/obj/vbelly/process()
+	var/all_done = TRUE //should it stop processing
+	var/should_update = FALSE //should it call a vore prefs ui update on mobs inside, and on the owner
+	switch(mode)
+		if (VORE_MODE_HOLD)
+			all_done = TRUE
+		if (VORE_MODE_DIGEST)
+			for (var/mob/living/prey in src)
+				if (!prey.check_vore_toggle(DIGESTABLE, VORE_MECHANICS_TOGGLES))
+					continue
+				prey.apply_damage(4, BURN)
+				if (prey.stat == DEAD)
+					var/pred_message = vore_replace(data[LIST_DIGEST_PRED], owner, prey, name)
+					var/prey_message = vore_replace(data[LIST_DIGEST_PREY], owner, prey, name)
+					vore_message(owner, pred_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					vore_message(prey, prey_message, SEE_OTHER_MESSAGES, warning=TRUE)
+					prey.release_belly_contents()
+					for (var/obj/item/item in prey)
+						if (!prey.dropItemToGround(item))
+							qdel(item)
+					if (prey.check_vore_toggle(LEAVE_ESSENCE_CUBE, VORE_MECHANICS_TOGGLES))
+						new /obj/item/essence_cube(src, prey)
```

Comment:
If SS13 mechanical death has occured, SS13 mechanics should be required to bring the body back.
https://github.com/Seris02/Skyrat-tg/pull/2
I've made a PR to implement this.